src/main/java/com/iexec/core/task/TaskService.java

```diff
@@ -105,6 +109,8 @@ public TaskService(
                 .orElseGet(() -> {
                         Task newTask = new Task(chainDealId, taskIndex, imageName,
```

Comment:
We could probably use a builder

Reply:
This needs modifications of the Task class (to generate the chainTaskId with the builder). So I'll leave as it is for now. We can change it in the future. Thanks anyway!